src/Datasource/EntityTrait.php

```diff
@@ -521,31 +522,10 @@ public function isEmpty(string $field): bool
                 $value === ''
             )
         ) {
-            return true;
+            return false;
         }
 
-        return false;
-    }
-
-    /**
-     * Checks that a field has a value.
-     *
-     * This method will return true for
-     *
-     * - Non-empty strings
-     * - Non-empty arrays
-     * - Any object
-     * - Integer, even `0`
-     * - Float, even 0.0
-     *
-     * and false in all other cases.
-     *
-     * @param string $field The field to check.
-     * @return bool
-     */
-    public function hasValue(string $field): bool
-    {
-        return !$this->isEmpty($field);
+        return true;
```

Comment:
How will the deprecation for isempty() work if hasValue calls isempty()?  Perhaps we'll need to clone the method body to be able to add the deprecation?

Reply:
We can sure that in in 5.next, jep.